### PR TITLE
Detect cycles when serializing log context

### LINKED ISSUE #19: Add cycle detection in Node sanitization for encoding
Hello!

I'm using Logtail and just noticed an error on trying to log some metadata based on objects in my app that have circular references. This is the stack trace:

```log
RangeError: Maximum call stack size exceeded
    at /my_app/node_modules/@logtail/node/src/node.ts:104:37
    at Array.forEach (<anonymous>)
    at Node.sanitizeForEncoding (/my_app/node_modules/@logtail/node/src/node.ts:104:29)
    at /my_app/node_modules/@logtail/node/src/node.ts:108:29
    at Array.forEach (<anonymous>)
    at Node.sanitizeForEncoding (/my_app/node_modules/@logtail/node/src/node.ts:104:29)
    at /my_app/node_modules/@logtail/node/src/node.ts:108:29
    at Array.forEach (<anonymous>)
    at Node.sanitizeForEncoding (/my_app/node_modules/@logtail/node/src/node.ts:104:29)
    at /my_app/node_modules/@logtail/node/src/node.ts:100:39
```

That links to this piece of code, which is ignoring possible reference cycles:

https://github.com/logtail/logtail-js/blob/174eee03ca3baebbddbad0cb17da23c11ea0fb93/packages/node/src/node.ts#L104-L112

So, if I may share my opinion, I believe cycles coming from the context object should be cut off by not propagating them after sanitization operation.

### PR DESCRIPTION
**Pull request by adikus**
Also stop before exceeding stack size on very nested logs

Fixes #19

This is a rebased version of https://github.com/logtail/logtail-js/pull/29, thanks @boly38 and @yagotome!

### CODE REVIEW

**Comment by adikus** [comment]
I'm hoping to have this released by tomorrow

**Review by yagotome — APPROVED** [review]
LGTM! Thanks @adikus for covering the circular reference case, that'll be helpful to me as a user of the lib 🙂

File: packages/node/src/node.ts [hunk 1]
```diff
@@ -102,20 +103,32 @@ export class Node extends Base {
 
       return value.toISOString();
     } else if (Array.isArray(value)) {
-      return value.map((item) => this.sanitizeForEncoding(item));
+      return value.map((item) => this.sanitizeForEncoding(item, maxDepth-1, visitedObjects));
```
Comment (yagotome) [inline packages/node/src/node.ts:106]: Arrays are also objects, and although unlikely they could also  be part of a circular reference. You could track them using the `visitedObject` set as well.

**Review by yagotome — CHANGES_REQUESTED** [review]
Just a minor edge case uncovered.

File: packages/node/src/node.ts (hunk 1 above)
Comment (adikus) [reply packages/node/src/node.ts:106]: Thanks, updated

**Review by boly38 — APPROVED** [review]
LGTM 👍

**Review by yagotome — APPROVED** [review]
Great stuff 👍🏼

**Comment by adikus** [comment]
Released as 0.1.15

**Pull request merged**